Zend/zend_inheritance.c

```diff
@@ -2320,6 +2325,15 @@ void zend_inheritance_check_override(const zend_class_entry *ce)
 		}
 	} ZEND_HASH_FOREACH_END();
 
+	ZEND_HASH_MAP_FOREACH_PTR(&ce->properties_info, p) {
+		if (p->flags & ZEND_ACC_OVERRIDE) {
+			zend_error_noreturn(
+				E_COMPILE_ERROR,
+				"%s::$%s has #[\\Override] attribute, but no matching parent property exists",
+				ZSTR_VAL(ce->name), zend_get_unmangled_property_name(p->name));
+		}
+	} ZEND_HASH_FOREACH_END();
```

Comment:
This can probably merged with the loop below (by removing the `ce->num_hooked_props` check).